tests/test_adv_parser.py

```diff
@@ -1737,3 +1737,33 @@ def test_parse_advertisement_data_relay_switch_1():
         rssi=-67,
         active=True,
     )
+
+
+def test_parse_advertisement_data_leak():
+    """Test parse_advertisement_data for the leak."""
+    ble_device = generate_ble_device("aa:bb:cc:dd:ee:ff", "any")
+    adv_data = generate_advertisement_data(
+        manufacturer_data={2409: b"\xc4407Lz\x18N\x98g^\x94Q<\x05\x00\x00\x00\x00"},
+        service_data={"0000fd3d-0000-1000-8000-00805f9b34fb": b"&\x00N"},
+        rssi=-72,
+    )
```

Comment:
Please add a test for passive scans as well.  In passive mode, the service_data will be empty